Xamarin.Forms.Platform.iOS/Renderers/TimePickerRenderer.cs

```diff
@@ -10,54 +11,83 @@ public class TimePickerRenderer : ViewRenderer<TimePicker, UITextField>
 	{
 		UIDatePicker _picker;
 		UIColor _defaultTextColor;
+		NoCaretField _noCaretField;
+
+		bool _disposed;
 
 		IElementController ElementController => Element as IElementController;
 
 		protected override void Dispose(bool disposing)
 		{
+			if (_disposed)
+				return;
+
 			if (disposing)
 			{
-				Control.EditingDidBegin -= OnStarted;
-				Control.EditingDidEnd -= OnEnded;
+				UnregisterEvents();
 
-				_picker.ValueChanged -= OnValueChanged;
+				if (_noCaretField != null)
+				{
+					_noCaretField.Dispose();
+					_noCaretField = null;
+				}
+
+				if (_picker != null)
+				{
+					_picker.Dispose();
+					_picker = null;
+				}
+
+				if (_defaultTextColor != null)
+				{
+					_defaultTextColor.Dispose();
+					_defaultTextColor = null;
+				}
 			}
 
+			_disposed = true;
+
 			base.Dispose(disposing);
 		}
 
 		protected override void OnElementChanged(ElementChangedEventArgs<TimePicker> e)
 		{
+			if (e.OldElement != null)
+			{
+				UnregisterEvents();
+			}
```

Comment:
On each platform, we're checking old element and setting `_noCaretField` (down below) for consistency.